Ymls

JaimeRZP opened the pull request.

damonge commented on input/ACTCMB.yml:
```diff
@@ -0,0 +1,69 @@
+tracers:
```
just a single section with the corresponding mapper

JaimeRZP replied:
Done!

damonge commented on input/ACTCMB.yml:
```diff
@@ -0,0 +1,69 @@
+    mapper_class: 'MapperACTCMB'
+    path_lite: 'data_ACTCMB'
+    
+  ACTCMB__BN:
```
only one mapper per yml file

JaimeRZP replied:
Done!

damonge commented on input/ACTCMB.yml:
```diff
@@ -0,0 +1,69 @@
+    mapper_class: 'MapperACTCMB'
+    path_lite: 'data_ACTCMB'
+
+cls:
```
And nothing else. throw everything else out

JaimeRZP replied:
Done!

carlosggarcia commented:
@JaimeRZP Thanks! Just had a very quick look. Two very minor comments: 
1. if I remember correctly, the mask for DESY1gc is binary, so applying the threshold at 0.5 is a bit meaningless (although it's what we did). Maybe should we set it to 0 to remind ourselves that it is binary? (@damonge what do you think) In any case, we should check that it is binary first.
2. Since the tracer names have to follow the convention of Name__bin (with 2 underscores), I think it would be good to do the same with the yml filenames as a way of reminding ourselves that that is the convention.

JaimeRZP commented on input/HSCDR1wl_0.yml:
```diff
@@ -0,0 +1,20 @@
+fnames_cosmos_ph:
+  - '/mnt/extraspace/damonge/Datasets/HSC_DR1/pdf-s17a_wide-9812.cat.fits'
+  - '/mnt/extraspace/damonge/Datasets/HSC_DR1/pdf-s17a_wide-9813.cat.fits'
+nbin_nz: 100
```
same as` fname_cosmos`, only grabbed from the config inside _get_nz()

damonge replied:
not sure I understand

JaimeRZP commented on input/HSCDR1wl_0.yml:
```diff
@@ -0,0 +1,20 @@
+  - '/mnt/extraspace/damonge/Datasets/HSC_DR1/pdf-s17a_wide-9812.cat.fits'
+  - '/mnt/extraspace/damonge/Datasets/HSC_DR1/pdf-s17a_wide-9813.cat.fits'
+nbin_nz: 100
+zlim_nz: [0.0, 4.0]
```
same as `fname_cosmos`, only grabbed from the config inside _get_nz()

damonge replied:
not sure I understand

JaimeRZP commented on input/HSCDR1wl_0.yml:
```diff
@@ -0,0 +1,20 @@
+nbin_nz: 100
+zlim_nz: [0.0, 4.0]
+mask_name: 'mask_HSC_wl0'
+mapper_class: 'MapperHSCDR1wl'
```
no `map_name:` ?

JaimeRZP replied:
Applies to all HSCwl

damonge replied:
nope

JaimeRZP commented on input/WIxSC_0.yml:
```diff
@@ -0,0 +1,23 @@
+mapper_class: MapperWIxSC
```
no `map_name`?

damonge replied:
no

JaimeRZP commented on input/WIxSC_0.yml:
```diff
@@ -0,0 +1,23 @@
+apply_galactic_correction: True
+mask: '/mnt/extraspace/damonge/Datasets/2MPZ_WIxSC/WISExSCOSmask.fits.gz'
+star_map: '/mnt/extraspace/damonge/Datasets/2MPZ_WIxSC/allwise_total_galactic.fits'
+use_halo_model: True
```
same comments as 2MPZ about `hod_params` and `use_halo_model`

JaimeRZP replied:
Applies to all WIxSC

damonge replied:
same answers

JaimeRZP commented on input/WIxSC_0.yml:
```diff
@@ -0,0 +1,23 @@
+bin_name: 'z0p10_0p15'
+nside_nl_threshold: 8129
+lmin_nl_from_data: 8192
+nl_analytic: True
```
maybe make nl_analytic an object in __init__ for clarity?

damonge replied:
Not sure what you mean by "an object in init"

JaimeRZP commented on input/WIxSC_0.yml:
```diff
@@ -0,0 +1,23 @@
+path_rerun: '/mnt/extraspace/damonge/Datasets/2MPZ_WIxSC/xcell_runs'
+coords: G
+z_edges: [0.10, 0.15]
+bin_name: 'z0p10_0p15'
```
maybe enumerate the possible bin names in the mapper for user clarity?

damonge replied:
The trouble is that these N(z)s will be saved to rerun files. So if we change the definition of the bin edges, we run the risk of reading the rerun files from the previous run with different bins, rather than recalculating them. This ensures that all distinct bins will be saved in different files.

JaimeRZP commented on input/WIxSC_0.yml:
```diff
@@ -0,0 +1,23 @@
+lmin_nl_from_data: 8192
+nl_analytic: True
+data_catalog: '/mnt/extraspace/damonge/Datasets/2MPZ_WIxSC/WIxSC.fits'
+spec_sample: '/mnt/extraspace/damonge/Datasets/2MPZ_WIxSC/zSpec-comp-WIxSC.csv'
```
not defined in __init__

damonge replied:
don't understand

JaimeRZP commented on input/eBOSS_ELG.yml:
```diff
@@ -0,0 +1,16 @@
+z_edges: [0.6, 1.1]
+path_rerun: '/mnt/extraspace/damonge/Datasets/eBOSS/xcell_runs'
+mask_name: 'mask_ELG'
+mapper_class: 'MappereBOSS'
```
no `map_name`, no `coords`?

damonge replied:
no

JaimeRZP commented on input/eBOSS_QSO_1.yml:
```diff
@@ -0,0 +1,16 @@
+random_catalogs:
+  - '/mnt/extraspace/damonge/Datasets/eBOSS/eBOSS_QSO_clustering_random-NGC-vDR16.fits'
+  - '/mnt/extraspace/damonge/Datasets/eBOSS/eBOSS_QSO_clustering_random-SGC-vDR16.fits'
+z_edges: [1.5, 4.0]
```
outside default z_edges in mapper

damonge replied:
I don't understand

JaimeRZP replied:
the range of the default z_edges only goes to z=3. Should we change the default in the mapper?

damonge replied:
But that doesn't really matter, right? This bin is meant to capture all QSOs beyond z=1.5, so this is just makes sure we get all of them.
I made it [1.5, 4] because that's actually the redshift range we used in Carlos' paper

JaimeRZP replied:
Ok!

JaimeRZP reviewed (commented):
General comments:
- We need to standarize `coords`, `map_name` and `path_rerun` across all ymls.
- I would  declare every neccesary config item in the __init__ of the mappers. This is generally the case but 2MPZ and WIxSC have configuration options only called in private functions. I feel like this will make users struggle to figure out what must go into their yml file. My idea is that all the yml file should do is fill the config objects mentioned in the __init__ of the mappers.

damonge commented:
Responding to @carlosggarcia first:

>if I remember correctly, the mask for DESY1gc is binary, so applying the threshold at 0.5 is a bit meaningless (although it's what we did). Maybe should we set it to 0 to remind ourselves that it is binary? (@damonge what do you think) In any case, we should check that it is binary first.

The original mask is binary, but that won't be the case after degrading

>Since the tracer names have to follow the convention of Name__bin (with 2 underscores), I think it would be good to do the same with the yml filenames as a way of reminding ourselves that that is the convention.

I'm not sure we should be so orthodox. There are mappers like the CIB where you have 3 frequency maps (353, 545 and 857 GHz), and I think it'd be best to call those "CIB_353" etc. for clarity

damonge commented on input/2MPZ.yml:
```diff
@@ -0,0 +1,16 @@
+mapper_class: Mapper2MPZ
+mask_name: mask_2MPZWISC
```
`map_name` is not a required key for all mappers, just some of them (actually, some of the ones you created)

damonge replied:
Actually, I think we only use this for ACT

damonge commented on input/2MPZ.yml:
```diff
@@ -0,0 +1,16 @@
+data_catalog: '/mnt/extraspace/damonge/Datasets/2MPZ_WIxSC/2MPZ.fits'
+n_jk_dir: 100
+mask: '/mnt/extraspace/damonge/Datasets/2MPZ_WIxSC/WISExSCOSmask.fits.gz'
+use_halo_model: True
```
But it is an attribute of the mapper: https://github.com/xC-ell/xCell/blob/11eebb08fb6d31a97032e9610e5479745afe08fa/xcell/cls/theory.py#L113

I.e. we can only use the halo model with certain mappers, so it's not something that can be applied to all mappers at once

damonge commented on input/2MPZ.yml:
```diff
@@ -0,0 +1,16 @@
+mask_name: mask_2MPZWISC
+path_rerun: '/mnt/extraspace/damonge/Datasets/2MPZ_WIxSC/xcell_runs'
+coords: G
+z_edges: [0.0, 0.1]
```
Agreed, but I think we should use this range for our master run. Basically, WIxSC covers the range z>0.1, so we don't gain anything by extending 2MPZ beyond 0.1. Of course, users can do whatever they want if anyone ever uses the code to get their own C_ells.

damonge commented on input/2MPZ.yml:
```diff
@@ -0,0 +1,16 @@
+n_jk_dir: 100
+mask: '/mnt/extraspace/damonge/Datasets/2MPZ_WIxSC/WISExSCOSmask.fits.gz'
+use_halo_model: True
+hod_params:
```
I'm not sure I follow

damonge commented on input/BOSS_CMASS.yml:
```diff
@@ -0,0 +1,16 @@
+z_edges: [0.4, 0.8]
+path_rerun: '/mnt/extraspace/damonge/Datasets/BOSS/xcell_runs'
+mask_name: 'mask_CMASS'
+mapper_class: 'MapperBOSS'
```
See responses above

damonge commented on input/BOSS_LOWZ.yml:
```diff
@@ -0,0 +1,16 @@
+z_edges: [0.1, 0.43]
+path_rerun: '/mnt/extraspace/damonge/Datasets/BOSS/xcell_runs'
+mask_name: 'mask_LOWZ'
+mapper_class: 'MapperBOSS'
```
See responses above

damonge commented on input/DELS_0.yml:
```diff
@@ -0,0 +1,12 @@
+completeness_map: '/mnt/extraspace/damonge/Datasets/DELS/Legacy_footprint_completeness_mask_128.fits'
+star_map: '/mnt/extraspace/damonge/Datasets/DELS/allwise_total_rot_1024.fits'
+mask_name: 'mask_DELS_decm36'
+mapper_class: 'MapperDELS'
```
- map_name: only ACT uses `map_name` as a required key
- `coords` goes in the global part of the yml file (i.e. it's not a per-mapper value)
- `path_rerun`: you're absolutely right, I'll add that

damonge replied:
Regarding masks: we should only name masks differently if they are actually different! xcell uses the mask names to figure out which distinct mode-coupling coefficients it needs to compute. If we name all masks differently (even if they are the same), we'll end up computing all possible mode coupling coefficients, which is super slow and disk-space consuming.

damonge replied:
Added path_rerun

damonge commented on input/DESY1gc_0.yml:
```diff
@@ -0,0 +1,8 @@
+zbin: 0
+mask_threshold: 0.5
+data_catalog: '/mnt/extraspace/damonge/Datasets/DES_Y1/redmagic_catalog/DES_Y1A1_3x2pt_redMaGiC_zerr_CATALOG.fits'
+file_mask: '/mnt/extraspace/damonge/Datasets/DES_Y1/redmagic_catalog/DES_Y1A1_3x2pt_redMaGiC_MASK_HPIX4096RING.fits'
```
Nope, see above.

damonge commented on input/DESY1wl_0.yml:
```diff
@@ -0,0 +1,7 @@
+file_nz: '/mnt/extraspace/damonge/Datasets/DES_Y1/shear_catalog/y1_redshift_distributions_v1.fits'
+path_rerun: '/mnt/extraspace/damonge/Datasets/DES_Y1/xcell_reruns/'
+mask_name: 'mask_DESY1wl0'
+mapper_class: 'MapperDESY1wl'
```
I think `path_rerun` is there, right?

I'll add `mode: shear`, although that's the default.

damonge replied:
Added `mode: shear`

damonge commented on input/HSCDR1wl_0.yml:
```diff
@@ -0,0 +1,20 @@
+depth_cut: 24.5
+z_edges: [0.3, 0.6]
```
By construction? In this case there's really no need. For DESY1 this is necessary because the bin assignment is not as simple, but in this case it is actually as simple as binning in photo-z.

damonge commented on input/HSCDR1wl_0.yml:
```diff
@@ -0,0 +1,20 @@
+     '/mnt/extraspace/damonge/Datasets/HSC_DR1/HSC_WIDE_VVDS_part2.fits']
+  - ['/mnt/extraspace/damonge/Datasets/HSC_DR1/HSC_WIDE_WIDE12H.fits']
+  - ['/mnt/extraspace/damonge/Datasets/HSC_DR1/HSC_WIDE_XMM.fits']
+fname_cosmos: '/mnt/extraspace/damonge/Datasets/HSC_DR1/Afterburner_reweighted_COSMOS_photoz_FDFC.fits'
```
I don't understand this.

damonge commented on input/ROSAT.yml:
```diff
@@ -1,61 +1,7 @@
+exposure_min: 100.0
+external_mask: '/mnt/extraspace/damonge/Datasets/2MPZ_WIxSC/WISExSCOSmask_equatorial.fits.gz'
+mask_name: 'mask_ROSAT'
+mapper_class: 'MapperROSATXray'
```
no need for any of those (including path_rerun)

damonge commented:
Now responding to @JaimeRZP 's global comments

>  We need to standarize coords, map_name and path_rerun across all ymls.

`coords` is a global option that we set in the `sphere` part of the yml file (i.e. not per mapper).

`map_name` is only used in the ACT mappers, but not required anywhere else.

`path_rerun`: only some mappers use this (those where saving some intermediate time-consuming data products is a good thing to do). By default `path_rerun` will be `None`.

>    I would declare every neccesary config item in the init of the mappers. This is generally the case but 2MPZ and WIxSC have configuration options only called in private functions. I feel like this will make users struggle to figure out what must go into their yml file. My idea is that all the yml file should do is fill the config objects mentioned in the init of the mappers.

I think this is what documentation is for (which should be the next step). The docstrings for each mapper should contain a description of all the yml attributes that can be associated to it.

carlosggarcia commented:
> I'm not sure we should be so orthodox. There are mappers like the CIB where you have 3 frequency maps (353, 545 and 857 GHz), and I think it'd be best to call those "CIB_353" etc. for clarity

@damonge In that case, the tracers should be called CIB__353 inside the yaml file too. We use the double underscore to define the "tracer barename" https://github.com/xC-ell/xCell/blob/11eebb08fb6d31a97032e9610e5479745afe08fa/xcell/cls/data.py#L112. This is used later to define which cls you want; e.g. https://github.com/xC-ell/xCell/blob/11eebb08fb6d31a97032e9610e5479745afe08fa/input/DESY1_eBOSS_P18CMBk_512.yml#L141-L144

damonge commented:
No, no, exactly: I would call this tracer `CIB__0` in the master yml file. But the individual yml file I'd call `CIB_353.yml`

carlosggarcia commented:
I think the code should work if you call it `CIB__343` inside the yml, as well. That might be clearer, no?

carlosggarcia commented:
I think so

damonge commented on input/ACTCMB_BN.yml:
```diff
@@ -0,0 +1,8 @@
+file_map: /mnt/extraspace/damonge/Datasets/ACT_DR4/compsep_maps/tilec_single_tile_BN_cmb_map_v1.2.0_joint.fits
+file_mask: /mnt/extraspace/damonge/Datasets/ACT_DR4/masks/compsep_masks/act_dr4.01_s14s15_BN_compsep_mask.fits
+map_name: ACT_CMB_BN
+lmax: 6000
```
Should there be a `beam_info` section for this mapper?

JaimeRZP replied:
Yes and in all ACT CMB and tSZ

damonge commented on input/ACTCMB_D56.yml:
```diff
@@ -0,0 +1,8 @@
+file_map: /mnt/extraspace/damonge/Datasets/ACT_DR4/compsep_maps/tilec_single_tile_D56_cmb_map_v1.2.0_joint.fits
+file_mask: /mnt/extraspace/damonge/Datasets/ACT_DR4/masks/compsep_masks/act_dr4.01_s14s15_D56_compsep_mask.fits
+map_name: ACT_CMB_D56
+lmax: 6000
```
Should there be a `beam_info` section for this mapper?

JaimeRZP replied:
Yes

damonge commented on input/ACTtSZ_BN.yml:
```diff
@@ -0,0 +1,8 @@
+file_map: /mnt/extraspace/damonge/Datasets/ACT_DR4/compsep_maps/tilec_single_tile_BN_comptony_map_v1.2.0_joint.fits
+file_mask: /mnt/extraspace/damonge/Datasets/ACT_DR4/masks/compsep_masks/act_dr4.01_s14s15_BN_compsep_mask.fits
+map_name: ACT_tSZ_BN
+lmax: 6000
```
Should there be a `beam_info` section for this mapper?

JaimeRZP replied:
yes

damonge commented on input/ACTtSZ_D56.yml:
```diff
@@ -0,0 +1,8 @@
+file_map: /mnt/extraspace/damonge/Datasets/ACT_DR4/compsep_maps/tilec_single_tile_D56_comptony_map_v1.2.0_joint.fits
+file_mask: /mnt/extraspace/damonge/Datasets/ACT_DR4/masks/compsep_masks/act_dr4.01_s14s15_D56_compsep_mask.fits
+map_name: ACT_tSZ_BN
+lmax: 6000
```
Should there be a `beam_info` section for this mapper?

JaimeRZP replied:
yes

damonge commented on input/SPT.yml:
```diff
@@ -0,0 +1,13 @@
+coords: G
+custom_auto: True
+mask_name: mask_SPT
+path_rerun: '/mnt/extraspace/damonge/Datasets/SPT/xcell_runs'
```
Should there be a `beam_info` section for this mapper?

JaimeRZP replied:
yes you are right! Added!

damonge commented on input/CIB_Lenz.yml:
```diff
@@ -0,0 +1,17 @@
+file_hm1: /mnt/extraspace/damonge/Datasets/CIB_Lenz/cib_evenring.hpx.fits
+file_hm2: /mnt/extraspace/damonge/Datasets/CIB_Lenz/cib_oddring.hpx.fits
+file_mask: /mnt/extraspace/damonge/Datasets/CIB_Lenz/mask_apod.hpx.fits
+gal_mask_mode: '0.2'
```
Humm, why do we need this? Isn't the mask defined in a single healpix map?

JaimeRZP replied:
You are right, mind fart

JaimeRZP replied:
you are right! corrected!

damonge commented on input/CIB_Lenz.yml:
```diff
@@ -0,0 +1,17 @@
+file_hm2: /mnt/extraspace/damonge/Datasets/CIB_Lenz/cib_oddring.hpx.fits
+file_mask: /mnt/extraspace/damonge/Datasets/CIB_Lenz/mask_apod.hpx.fits
+gal_mask_mode: '0.2'
+sp_mask_mode:  '545'
```
Same as above

JaimeRZP replied:
Done!

JaimeRZP replied:
Done!

damonge commented on input/CIB_Lenz.yml:
```diff
@@ -0,0 +1,17 @@
+mapper_class: MapperCIBLenz
```
We actually have 3 different frequencies for this map, so we should create 3 different files. Have a look at these ones I created for Baptiste's project:
```
/mnt/extraspace/damonge/Datasets/CIB_Lenz/map_353_2.5_20gp.yml
/mnt/extraspace/damonge/Datasets/CIB_Lenz/map_545_2.5_20gp.yml
/mnt/extraspace/damonge/Datasets/CIB_Lenz/map_857_2.5_20gp.yml
```

JaimeRZP replied:
Generated three files now!

damonge commented on input/CIB_Lenz.yml:
```diff
@@ -0,0 +1,17 @@
+             'nside_native': 2048,
+             'nside_wanted': 4096}]
+coords: G
+custom_auto: True
```
Isn't this hard-coded into anything that inherits from `PlankBase`?
https://github.com/xC-ell/xCell/blob/11eebb08fb6d31a97032e9610e5479745afe08fa/xcell/mappers/mapper_Planck_base.py#L28

JaimeRZP replied:
you are right, removing custom_auto: True from all of Planck_base children

damonge commented on input/CIB_Lenz.yml:
```diff
@@ -0,0 +1,17 @@
+             'nside_wanted': 4096}]
+coords: G
+custom_auto: True
+mask_name: mask_CIB_Lenz 
```
note that the mask name will be different for different frequencies

JaimeRZP replied:
Done!

damonge commented on input/NVSS.yml:
```diff
@@ -0,0 +1,9 @@
+mapper_class: MapperNVSS
+map_name: NVSS
```
not needed

JaimeRZP replied:
Done!

JaimeRZP replied:
Done!

damonge commented on input/P15tSZ.yml:
```diff
@@ -0,0 +1,12 @@
+file_map: /mnt/extraspace/damonge/Datasets/P15tSZ/COM_CompMap_Compton-SZMap-milca-ymaps_2048_R2.00.fits
+file_noise: /mnt/extraspace/damonge/Datasets/P15tSZ/COM_CompMap_Compton-SZMap-milca-stddev_2048_R2.00.fits
+file_mask: /mnt/extraspace/damonge/Datasets/P15tSZ/COM_CompMap_Compton-SZMap-masks_2048_R2.01.fits
+gp_mask_mode: '0.5'
```
I've normally used 0.6 in the past

JaimeRZP replied:
Changed!

damonge commented on input/P15tSZ.yml:
```diff
@@ -0,0 +1,12 @@
+map_name: P15tSZ
+file_map: /mnt/extraspace/damonge/Datasets/P15tSZ/COM_CompMap_Compton-SZMap-milca-ymaps_2048_R2.00.fits
+file_noise: /mnt/extraspace/damonge/Datasets/P15tSZ/COM_CompMap_Compton-SZMap-milca-stddev_2048_R2.00.fits
+file_mask: /mnt/extraspace/damonge/Datasets/P15tSZ/COM_CompMap_Compton-SZMap-masks_2048_R2.01.fits
```
These paths are wrong. Should be `/mnt/extraspace/damonge/Datasets/Planck_tSZ/COM_CompMap_YSZ_R2.02/`

damonge replied:
Also, shouldn't there be a `ps_mask` and a `gp_mask`?

JaimeRZP replied:
you are right but they are the same file

JaimeRZP replied:
Also updated the paths.

damonge commented on input/P15tSZ.yml:
```diff
@@ -0,0 +1,12 @@
+mapper_class: MapperP15tSZ
+map_name: P15tSZ
+file_map: /mnt/extraspace/damonge/Datasets/P15tSZ/COM_CompMap_Compton-SZMap-milca-ymaps_2048_R2.00.fits
+file_noise: /mnt/extraspace/damonge/Datasets/P15tSZ/COM_CompMap_Compton-SZMap-milca-stddev_2048_R2.00.fits
```
I don't think this attribute exists for the Planck tSZ map

JaimeRZP replied:
Removed!

damonge commented on input/P18CMBK.yml:
```diff
@@ -0,0 +1,11 @@
+mapper_class: MapperP18CMBK
+map_name: mask_P18CMB_kappa
```
not needed

JaimeRZP replied:
Done!

damonge commented on input/P18CMBK.yml:
```diff
@@ -0,0 +1,11 @@
+coords: G
+custom_auto: True
+mask_name: P18CMB_kappa
+path_rerun: '/mnt/extraspace/damonge/Datasets/P18CMBK/xcell_runs'
```
Don't think this path exists

JaimeRZP replied:
Solved!

damonge commented on input/P15tSZ.yml:
```diff
@@ -0,0 +1,12 @@
+coords: G
+custom_auto: True
+mask_name: mask_P15tSZ
+path_rerun: '/mnt/extraspace/damonge/Datasets/P15tSZ/xcell_runs'
```
Don't think this path exists

JaimeRZP replied:
The folder doesn't exist but I have now pointed it out to a more apropiate path. Would we need to create the rerun folder in advance or would it be created after the run if it doesn't exist yet?

damonge replied:
This happens within the code, but if the preceding folder doesn't exist, an error will be thrown

damonge commented on input/P15tSZ.yml:
```diff
@@ -0,0 +1,12 @@
+ps_mask_mode: ['default']
+beam_info: [{'type': 'Gaussian', 'FWHM_arcmin': 10.0}]
+coords: G
+custom_auto: True
```
Not needed. All mappers inheriting from `PlanckBase` have this hard-coded: https://github.com/xC-ell/xCell/blob/11eebb08fb6d31a97032e9610e5479745afe08fa/xcell/mappers/mapper_Planck_base.py#L28

JaimeRZP replied:
Done!

damonge commented on input/P18SMICA.yml:
```diff
@@ -0,0 +1,13 @@
+beam_info: [{'type': 'Gaussian', 'FWHM_arcmin': 5.0}])
+gp_mask_mode: '0.6'
+coords: G
+custom_auto: True
```
Not needed

JaimeRZP replied:
Done!

damonge commented on input/P18SMICA.yml:
```diff
@@ -0,0 +1,13 @@
+file_gp_mask: /mnt/extraspace/damonge/Datasets/P18SMICA_NOSZ/HFI_Mask_GalPlane-apo2_2048_R2.00.fits
+file_sp_mask: /mnt/extraspace/damonge/Datasets/P18SMICA_NOSZ/HFI_Mask_PointSrc_2048_R2.00.fits
+beam_info: [{'type': 'Gaussian', 'FWHM_arcmin': 5.0}])
+gp_mask_mode: '0.6'
```
Is there no point source mask?

JaimeRZP replied:
sp --> ps

damonge commented on input/SPT.yml:
```diff
@@ -0,0 +1,13 @@
+file_sp_mask: /mnt/extraspace/damonge/Datasets/SPT/sptsz_planck_ymap_healpix/SPTSZ_point_source_mask_nside_8192_binary_mask.fits
+ps_mask_mode: ['default']
+coords: G
+custom_auto: True
```
not needed

JaimeRZP replied:
Done!

JaimeRZP replied:
My bad! Removed it now!

damonge commented on input/SPT.yml:
```diff
@@ -0,0 +1,13 @@
+file_hm2: /mnt/extraspace/damonge/Datasets/SPT/sptsz_planck_ymap_healpix/SPTSZ_Planck_min_variance_ymap_half2.fits
+file_gp_mask: /mnt/extraspace/damonge/Datasets/SPT/sptsz_planck_ymap_healpix/SPTSZ_dust_mask_top_5percent.fits
+gp_mask_mode: 'default'
+file_sp_mask: /mnt/extraspace/damonge/Datasets/SPT/sptsz_planck_ymap_healpix/SPTSZ_point_source_mask_nside_8192_binary_mask.fits
```
I think this is `ps_mask`, not `sp_mask`

JaimeRZP replied:
sp --> ps

damonge replied:
it still reads `sp` above, right?

JaimeRZP replied:
Done!

damonge commented on input/P18SMICA.yml:
```diff
@@ -0,0 +1,13 @@
+file_hm1: /mnt/extraspace/damonge/Datasets/P18SMICA_NOSZ/COM_CMB_IQU-smica-nosz_2048_R3.00_hm1.fits
+file_hm2: /mnt/extraspace/damonge/Datasets/P18SMICA_NOSZ/COM_CMB_IQU-smica-nosz_2048_R3.00_hm2.fits
+file_gp_mask: /mnt/extraspace/damonge/Datasets/P18SMICA_NOSZ/HFI_Mask_GalPlane-apo2_2048_R2.00.fits
+file_sp_mask: /mnt/extraspace/damonge/Datasets/P18SMICA_NOSZ/HFI_Mask_PointSrc_2048_R2.00.fits
```
Also, it should be `ps_mask` not `sp_mask`

JaimeRZP replied:
Done! The rerun file doesn't exist but it would be created in the right directory.

damonge reviewed (changes requested):
@JaimeRZP thanks (and sorry for the many comments!). Main repeated ones are:
- `map_name`s are not needed in general. Same for `coords`.
- `custom_auto` is also not needed in the yml files.
- I think we need to be more careful with the path names and whether or not both gp and ps masks are needed for the Planck-like mappers.

damonge commented:
@carlosggarcia ok, then I agree with you that it's probably best if we put the double underscore in the yml file names. Remind us to do that when you make your review if we haven't done it.

JaimeRZP commented on input/ACTCMB_BN.yml:
```diff
@@ -0,0 +1,8 @@
+mapper_class: MapperACTCMB
+mask_name: mask_ACT_compsep_BN
+path_rerun: '/mnt/extraspace/damonge/Datasets/ACT_DR4/xcell_runs'
+coords: G
```
Done!

JaimeRZP commented on input/ACTCMB_D56.yml:
```diff
@@ -0,0 +1,8 @@
+mapper_class: MapperACTCMB
+mask_name: mask_ACT_compsep_D56
+path_rerun: '/mnt/extraspace/damonge/Datasets/ACT_DR4/xcell_runs'
+coords: G
```
Done!

JaimeRZP commented on input/ACTk_BN.yml:
```diff
@@ -0,0 +1,9 @@
+mapper_class: MapperACTk
+mask_name: mask_ACT_kappa_BN
+path_rerun: '/mnt/extraspace/damonge/Datasets/ACT_DR4/xcell_runs'
+coords: G
```
Done!

JaimeRZP commented on input/ACTk_D56.yml:
```diff
@@ -0,0 +1,9 @@
+mapper_class: MapperACTk
+mask_name: mask_ACT_kappa_D56
+path_rerun: '/mnt/extraspace/damonge/Datasets/ACT_DR4/xcell_runs'
+coords: G
```
Done!

JaimeRZP commented on input/ACTtSZ_BN.yml:
```diff
@@ -0,0 +1,8 @@
+mapper_class: MapperACTtSZ
+mask_name: mask_ACT_compsep_BN
+path_rerun: '/mnt/extraspace/damonge/Datasets/ACT_DR4/xcell_runs'
+coords: G
```
Done!

JaimeRZP commented on input/ACTtSZ_D56.yml:
```diff
@@ -0,0 +1,8 @@
+mapper_class: MapperACTtSZ
+mask_name: mask_ACT_compsep_D56
+path_rerun: '/mnt/extraspace/damonge/Datasets/ACT_DR4/xcell_runs'
+coords: G
```
Done!

JaimeRZP commented on input/NVSS.yml:
```diff
@@ -0,0 +1,9 @@
+file_mask: /mnt/extraspace/damonge/Datasets/NVSS/mask.fits
+redshift_catalog: /mnt/extraspace/damonge/Datasets/NVSS/100sqdeg_1uJy_s1400.fits
+mask_sources: /mnt/extraspace/damonge/Datasets/NVSS/source_masks_nvss.txt
+coords: G
```
Done!

JaimeRZP commented on input/CIB_Lenz.yml:
```diff
@@ -0,0 +1,17 @@
+             {'type': 'Pixel',
+             'nside_native': 2048,
+             'nside_wanted': 4096}]
+coords: G
```
Done!

JaimeRZP commented on input/P15tSZ.yml:
```diff
@@ -0,0 +1,12 @@
+gp_mask_mode: '0.5'
+ps_mask_mode: ['default']
+beam_info: [{'type': 'Gaussian', 'FWHM_arcmin': 10.0}]
+coords: G
```
Done!

JaimeRZP commented on input/P18CMBK.yml:
```diff
@@ -0,0 +1,11 @@
+file_noise: /mnt/extraspace/damonge/Datasets/Planck_data/COM_Lensing_4096_R3.00/MV/nlkk.dat 
+mask_aposize: 0.2
+mask_apotype: 'C1'
+coords: G
```
Done!

JaimeRZP commented on input/SPT.yml:
```diff
@@ -0,0 +1,13 @@
+gp_mask_mode: 'default'
+file_sp_mask: /mnt/extraspace/damonge/Datasets/SPT/sptsz_planck_ymap_healpix/SPTSZ_point_source_mask_nside_8192_binary_mask.fits
+ps_mask_mode: ['default']
+coords: G
```
Done!

JaimeRZP commented on input/P18SMICA.yml:
```diff
@@ -0,0 +1,13 @@
+file_sp_mask: /mnt/extraspace/damonge/Datasets/P18SMICA_NOSZ/HFI_Mask_PointSrc_2048_R2.00.fits
+beam_info: [{'type': 'Gaussian', 'FWHM_arcmin': 5.0}])
+gp_mask_mode: '0.6'
+coords: G
```
Done!

JaimeRZP commented on input/CIB_Lenz.yml:
```diff
@@ -0,0 +1,17 @@
+mapper_class: MapperCIBLenz
+map_name: CIB_Lenz
```
Done!

JaimeRZP replied:
Done!

JaimeRZP commented on input/NVSS.yml:
```diff
@@ -0,0 +1,9 @@
+mapper_class: MapperNVSS
+map_name: NVSS
+data_catalog: /mnt/extraspace/damonge/Datasets/NVSS/nvss.fits
+file_mask: /mnt/extraspace/damonge/Datasets/NVSS/mask.fits
```
Done!

JaimeRZP replied:
Done!

JaimeRZP commented on input/P15CIB.yml:
```diff
@@ -0,0 +1,13 @@
+mapper_class: MappereP15CIB
```
I mean it is already written

damonge replied:
Just read this. OK, I'll read the issue and get back to you.

damonge replied:
See below. I'm fine removing the code, but the tests need to go as well.

JaimeRZP commented on input/P18SMICA.yml:
```diff
@@ -0,0 +1,13 @@
+mapper_class: 'MapperP18SMICA'
+map_name: P18SMICA
```
Done!

JaimeRZP commented on input/P15tSZ.yml:
```diff
@@ -0,0 +1,12 @@
+mapper_class: MapperP15tSZ
+map_name: P15tSZ
```
Removed!

JaimeRZP commented on input/P18CMBK.yml:
```diff
@@ -0,0 +1,11 @@
+map_name: mask_P18CMB_kappa
+file_klm: /mnt/extraspace/damonge/Datasets/Planck_data/COM_Lensing_4096_R3.00/MV/dat_klm.fits
+file_mask: /mnt/extraspace/damonge/Datasets/Planck_data/COM_Lensing_4096_R3.00/mask.fits.gz
+file_noise: /mnt/extraspace/damonge/Datasets/Planck_data/COM_Lensing_4096_R3.00/MV/nlkk.dat 
```
Solved!

JaimeRZP commented on input/P18CMBK.yml:
```diff
@@ -0,0 +1,11 @@
+mask_aposize: 0.2
+mask_apotype: 'C1'
+coords: G
+custom_auto: True
```
Done!

JaimeRZP commented:
Just finished adressing David's feedback. Some quick remarks:

- Removed `map_name` across al .yml's
- Removed `custom_auto: True` from all `Planck_base` children mappers' .yml's
- Made a .yml for each freq of CIB_Lenz maps
- About rerun paths. Some of the paths I have written do not exist. Is that a problem? In other words, do they need to be created before the run or would the run create them when saving the files?

carlosggarcia commented:
@JaimeRZP the folders do not need to exists: https://github.com/xC-ell/xCell/blob/11eebb08fb6d31a97032e9610e5479745afe08fa/xcell/mappers/utils.py#L13

damonge commented on xcell/mappers/mapper_P15CIB.py:
```diff
@@ -1,56 +0,0 @@
-from .mapper_Planck_base import MapperPlanckBase
```
Bring it back! I just meant that we won't include it in the big run, but it's ok to have the code. We're also not including KV450 in the big run, but we have the code.

JaimeRZP replied:
Brought it back. I didn't mean to remove it in this PR sorry!

damonge commented on input/NVSS.yml:
```diff
@@ -0,0 +1,10 @@
+DEC_min_deg: -40
+flux_min_mJy: 10
+fux_max_mJy: 1000
+GLAT_max_deg: 5
```
Can we make this 10 instead of 5? Sorry, I know this is not what I used in the original yml

JaimeRZP replied:
Done!

damonge reviewed (commented):
Thanks a lot @JaimeRZP . A few follow-up comments. I'll look at the new issues and the Planck yml files again in more detail tomorrow.

damonge commented on input/CIB_Lenz_545.yml:
```diff
@@ -0,0 +1,14 @@
+    nside_native: 2048
+    nside_wanted: 4096
+mask_name: mask_CIB_Lenz_545 
+path_rerun: '/mnt/extraspace/damonge/Datasets/CIB_Lenz_353/xcell_runs'
```
545

JaimeRZP replied:
Solved!

damonge commented on input/P15tSZ.yml:
```diff
@@ -0,0 +1,11 @@
+  - type: 'Gaussian'
+    FWHM_arcmin: 10.0
+mask_name: mask_P15tSZ
+path_rerun: '/mnt/extraspace/damonge/Datasets/P15tSZ/COM_CompMap_YSZ_R2.02/xcell_runs'
```
Planck_tSZ

JaimeRZP replied:
Solved!

damonge commented on input/P18SMICA.yml:
```diff
@@ -0,0 +1,12 @@
+file_map: /mnt/extraspace/damonge/Datasets/P18SMICA_NOSZ/COM_CMB_IQU-smica-nosz_2048_R3.00_full.fits
+file_hm1: /mnt/extraspace/damonge/Datasets/P18SMICA_NOSZ/COM_CMB_IQU-smica-nosz_2048_R3.00_hm1.fits
+file_hm2: /mnt/extraspace/damonge/Datasets/P18SMICA_NOSZ/COM_CMB_IQU-smica-nosz_2048_R3.00_hm2.fits
+file_gp_mask: /mnt/extraspace/damonge/Datasets/P18SMICA_NOSZ/HFI_Mask_GalPlane-apo2_2048_R2.00.fits
```
In `Planck_masks`

JaimeRZP replied:
Solved!

damonge commented on input/P18SMICA.yml:
```diff
@@ -0,0 +1,12 @@
+beam_info: 
+  - type: 'Gaussian'
+    FWHM_arcmin': 5.0
+gp_mask_mode: '0.6'
```
```suggestion
gp_mask_mode: '0.6'
ps_mask_mode: ['F100', 'F143', 'F217', 'F353']
```

damonge replied:
And then, in the mapper code you need to add
```
 self.ps_mask_modes = {'F100': 0,
                                        'F143': 1,
                                        'F217': 2,
                                        'F353': 3}
self.ps_mask_mode = config.get('ps_mask_mode', ['F100', 'F143', 'F217', 'F353'])
```

damonge commented on input/P18SMICA.yml:
```diff
@@ -0,0 +1,12 @@
+    FWHM_arcmin': 5.0
+gp_mask_mode: '0.6'
+mask_name: mask_P18SMICA
+path_rerun: '/mnt/extraspace/damonge/Datasets/P18SMICA_NOSZ/xcell_runs'
```
This doesn't exist

JaimeRZP replied:
Solved!

damonge commented on input/P18SMICA.yml:
```diff
@@ -0,0 +1,12 @@
+mapper_class: 'MapperP18SMICA'
+file_map: /mnt/extraspace/damonge/Datasets/P18SMICA_NOSZ/COM_CMB_IQU-smica-nosz_2048_R3.00_full.fits
+file_hm1: /mnt/extraspace/damonge/Datasets/P18SMICA_NOSZ/COM_CMB_IQU-smica-nosz_2048_R3.00_hm1.fits
+file_hm2: /mnt/extraspace/damonge/Datasets/P18SMICA_NOSZ/COM_CMB_IQU-smica-nosz_2048_R3.00_hm2.fits
```
These paths don't exist. `Planck_SMICA`

JaimeRZP replied:
Solved!

damonge reviewed (changes requested):
Last few changes!

carlosggarcia reviewed (changes requested):
Quick review: I've pushed some changes (rename files to use `__` and some typos or easy fixes). It was fastest that way. I have also pushed a unit test to test the master.yml that will only run in Glamdring. 

Some notes regarding the yml files that you will know how to solve better:
- ACT: missing `map_name`
- WIxSC: mask doesn't exist and I am not sure what should be used
- If you can populate the remaining of the master.yml file it'd be great. That way we can use the `Data` class in the test.

JaimeRZP commented:
Hi @carlosggarcia! Just had a quick look at your comments:
1. Added names
2. Pointed the yml to the galactic mask
3.  I have gone over the master file. I have changed the order of some mappers to follow alphabetical so it is easier to check. However I haven't seen any yml missing... What is the file missing that you would like to me to populate?

carlosggarcia commented:
@JaimeRZP the `outdir`, the `sphere` section, etc.

carlosggarcia commented:
@damonge a short summary of what I think it's missing from a quick read. I can do it if you want:
- Fill the `master.yml` with the `recompute`, `sphere`, `output`, `bpw`, etc sections so we can add a test in the `test_input_master.py` with the Data class.
- Run the test and see that works. If so, I think I'm happy to merge.

damonge commented:
@carlosggarcia ok, thanks a lot. And what did we want this test to do? Just generate low-res masks? Something else?

carlosggarcia commented:
yes, the test is already written. It basically loops over the tracers and request nside 32 maps and masks

damonge commented:
I'm going to unilaterally merge this one. I've checked that the input yaml files can now be used to generate all masks and maps in Galactic and Equatorial coordinates. The rest of the master yaml file can be addressed in the other open PRs.
